src/hooks/useDownload.ts

```diff
@@ -39,14 +39,24 @@ async function internal<T>({ src, dst, headers, ephemeral, logger }: DownloadOpt
     return () => memo ?? (memo = hash_key(src))
   })()
   const mtime_entry = () => hash().join("mtime")
+  const etag_entry = () => hash().join("etag")
 
   const { numpty } = useFlags()
   dst ??= hash().join(src.path().basename())
   if (src.protocol === "file:") throw new Error()
 
-  if (!ephemeral && mtime_entry().isFile() && dst.isReadableFile()) {
+  if (!ephemeral && dst.isReadableFile()) {
+    
     headers ??= {}
-    headers["If-Modified-Since"] = await mtime_entry().read()
+    if(mtime_entry().isFile() && etag_entry().isFile()){
```

Comment:
https://stackoverflow.com/a/10671877

Don’t do both. Please research the task at hand before attempting it.

Reply:
Sure... I will. Thanks for the clarification

Reply:
Fixed